### PR TITLE
Agent installation docs

### PR DESCRIPTION
**Pull request by charlieegan3**
Fixes https://github.com/jetstack/preflight/issues/127

This is a basic write up of how to install the agent. I think it might also be worth including an example YAML for gathering GKE data?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -121,26 +126,142 @@ packages](./docs/how_to_write_packages.md).
...
+
+### Agent
+
+The following instructions walk through the installation of the Preflight agent
+to gather data about cluster pods and send them to the backend for analysis.
+
+**To complete the secret manifest below, you will need to have a Preflight
+token.**
+
+First create a namespace for the preflight components:
```
Comment (wwwil) [inline README.md]: I feel like these manifests should be in files in `examples/` or `deploy/` so they are easier for user to consume, though this might be an issue for a future PR.

File: README.md [hunk 2]
```diff
@@ -121,26 +126,142 @@ packages](./docs/how_to_write_packages.md).
...
+        resources:
+          requests:
+            memory: "200Mi"
+            cpu: "200m"
+          limits:
+            memory: "200Mi"
+            cpu: "200m"
+```
+
+### Preflight V1 (check)
```
Comment (wwwil) [inline README.md]: Maybe to make things cleaner all the info about check should be moved out to another file in `docs/` and just mentioned and linked once above. Might also be best for another PR.

Comment (charlieegan3) [reply README.md]: I can move them out if that's the popular vote.

My reasoning here was that we have documentation in a lot of places - this is becoming harder to keep in sync as we deprecate check. 

Until then, I wanted to keep the changes minimal and quite self contained until we're ready to remove check references completely...

File: README.md (hunk 1 above)
Comment (charlieegan3) [reply README.md]: I think my other comment touches on this.

File: README.md (hunk 2 above)
Comment (wwwil) [reply README.md]: Yeah I can see the reasoning there. I think I just don't like the cluttered feel in the mean time but I can always just get over it.

My thinking was that we actually move all check specific stuff into `docs/check.md`, refer to this once in `README.md` and otherwise everything can focus on the agent and new SaaS way of working.

Comment (charlieegan3) [reply README.md]: Sounds reasonable, I'll see what that looks like.

Comment (charlieegan3) [reply README.md]: ffe6c83351954fec31df77d8c638f3584b4fbcb7 is an attempt at this.

Comment (charlieegan3) [reply README.md]: I've tried to move all the check docs and examples to one place - leaving the new readme as a clear process for installing the agent.

File: README.md [hunk 3]
```diff
@@ -18,29 +18,22 @@ checks using [Open Policy Agent (OPA)](https://www.openpolicyagent.org/).
 **[Preflight _agent_](#agent).**
 While this is happening the docs may be inconsistent and the repo structure may
 change.
+[Docs for this component can still be found here](./docs/installation_manual_in_cluster.md).
```
Comment (wwwil) [inline README.md]:
```suggestion
[Docs for the check tool can still be found here](./docs/installation_manual_in_cluster.md).
```

Just to clarify as we mentioned two components in the previous sentence.

Comment (wwwil) [inline README.md]:
```suggestion
[Docs for this component can still be found here](./docs/check.md).
```

File: docs/check.md [hunk 4]
```diff
@@ -0,0 +1,439 @@
...
+:warning: :construction:
+
+The Preflight _check_ tool also uses _data gatherers_ to collect required data,
+but unlike the _agent_ it evaluates this data and produces a report locally.
+
+**This functionality is now deprecated in favour of the _agent_.** Previous
+versions of the _check_ tool can still be downloaded and used, however it is no
+longer being maintained and will be removed from this repository.
+
+## Packages
```
Comment (wwwil) [inline docs/check.md]: The packages are still in use so I think this section should either stay in `README.md` or move to `docs/packages.md`

Comment (charlieegan3) [reply docs/check.md]: But they're not in use by users of the agent - and will likely go before they come back from the public repo.

File: docs/check.md [hunk 5]
```diff
@@ -0,0 +1,439 @@
...
+Create your `preflight.yaml` configuration file. There is full [configuration
+documentation](./docs/configuration.md) available, as well as several example
+files in [`./examples`](./examples).
+
+### Locally
+
+By default Preflight looks for a configuration at `./preflight.yaml`. Once this
+is set up, run a Preflight check like so:
+
+``` preflight check ```
```
Comment (wwwil) [inline docs/check.md]:
```suggestion
``` 
preflight check
```
```

File: README.md (hunk 3 above)
Comment (charlieegan3) [reply README.md]: fixed in aff5c9841c73a6736816db01edf4552ed1edf1df

Comment (charlieegan3) [reply README.md]: fixed in aff5c9841c73a6736816db01edf4552ed1edf1df

File: docs/check.md (hunk 4 above)
Comment (wwwil) [reply docs/check.md]: Ah yeah! I might PR this paragraph back into the private repo's `README.md` then.

Comment (wwwil) [reply docs/check.md]: Actually I think it would still be good to briefly mention packages. Maybe just:

```
## Packages

Policies for cluster configuration are encoded into *Preflight packages*. 
Each package focuses on a different aspect of the cluster, for example the
`gke` package provides rules for the configuration of a GKE cluster.

Preflight packages are implemented using
[Open Policy Agent](https://www.openpolicyagent.org) with evaluation
taking place in the SaaS backend.
```

Comment (charlieegan3) [reply docs/check.md]: Yeah, good idea, added in: 835d92fd9e963624e9bbe8ec67839624424578d0
